### PR TITLE
[Component with its own class name as service type] LPS-197867 Refactor components from dynamic-data-mapping modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-197867

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:forms

**Comment by catarineaguiar** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
no regressions found

**Review by rafaprax — APPROVED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/helper/DDMSearchUtil.java [hunk 1]
```diff
@@ -63,10 +59,11 @@ public SearchContext buildStructureLayoutSearchContext(
 		return searchContext;
 	}
 
-	public SearchContext buildStructureSearchContext(
+	public static SearchContext buildStructureSearchContext(
 		long companyId, long[] groupIds, long userId, long classNameId,
 		Long classPK, String name, String description, String storageType,
 		Integer type, int status, int start, int end,
+		DDMPermissionSupport ddmPermissionSupport,
```
Comment (rafaprax) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/helper/DDMSearchUtil.java:66]: Hey @MrJohn1911, just pass the result of the `ddmPermissionSupport.getStructureModelResourceName(classNameId)` if possible. Thx

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/helper/DDMSearchUtil.java [hunk 2]
```diff
@@ -153,7 +157,7 @@ public SearchContext buildTemplateSearchContext(
 		try {
 			searchContext.setAttribute(
 				"resourcePermissionName",
-				_ddmPermissionSupport.getTemplateModelResourceName(
+				ddmPermissionSupport.getTemplateModelResourceName(
```
Comment (rafaprax) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/helper/DDMSearchUtil.java:160]: Hey @MrJohn1911, just pass the result of the `ddmPermissionSupport. getTemplateModelResourceName(resourceClassNameId)` if possible. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by rafaprax** [comment]
Hey @MrJohn1911, see the above comments and resend. Thx

**Pull request closed without merging**